Python/formatter_unicode.c

```diff
@@ -746,6 +800,15 @@ get_locale_info(enum LocaleType type, LocaleInfo *locale_info)
         locale_info->grouping = no_grouping;
         break;
     }
+    if (frac_type == LT_UNDERSCORE_LOCALE) {
+        locale_info->frac_thousands_sep = PyUnicode_FromOrdinal('_');
```

Comment:
PyUnicode_FromOrdinal() can fail. You need something like `if (!locale_info->frac_thousands_sep) return -1;`.